src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -110,5 +138,69 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
 
             return operand;
         }
+
+        private static string NormalizeCollectionItems(string bracketLiteralText, NormalizeFunction normalizeFunc)
+        {
+            string[] items = bracketLiteralText.Substring(1, bracketLiteralText.Length - 2).Split(',')
+                .Select(s => s.Trim()).ToArray();
+
+            StringBuilder builder = new StringBuilder();
+            for (int i = 0; i < items.Length; i++)
+            {
+                string convertedItem = normalizeFunc(items[i]);
+                if (i != items.Length - 1)
+                {
+                    builder.AppendFormat(CultureInfo.InvariantCulture, "{0},", convertedItem);
+                }
+                else
+                {
+                    // No trailing comma separator for last str of the collection.
+                    builder.Append(convertedItem);
+                }
+            }
+
+            return String.Format(CultureInfo.InvariantCulture, "[{0}]", builder.ToString());
+        }
+
+        /// <summary>
+        /// Function to normalize quoted string, ensuring single quotes are escaped properly.
+        /// If the string is double-quoted, no op since single quote doesn't need to be escaped.
+        /// </summary>
+        /// <param name="str">The quoted string item to be normalized.</param>
+        /// <returns>The double-quoted string with single quotes properly escaped.</returns>
+        private static string NormalizeStringItem(string str)
+        {
+            // Skip conversion if the items are already in double-quote format (for backward compatibility).
+            // Note that per ABNF, query option strings should use single quotes.
+            Debug.Assert(str[0] == '\'' || str[0] == '"', "string needs to be single/double quoted.");

```

Comment:
Should this be an OData error instead of a Debug.Assert?

Reply:
This should be a Debug.Assert since the expected type is string, therefore the value in the json should be single-quote or double-quote.